nservicebus/outbox/index.md

```diff
@@ -79,8 +79,61 @@ In order to gradually convert an entire system from the DTC to the Outbox:
 WARNING: When verifying outbox functionality, it can sometimes be helpful to temporarily [stop the MSDTC Service](https://technet.microsoft.com/en-us/library/cc770732.aspx). This ensures that the Outbox is working as expected, and no other resources are enlisting in distributed transactions.
 
 
+## Message identity and idempotent processing
+
+If handlers are idempotent then outbox might not be required at all. When considering not using the outbox and implementing idempotent processing keep in mind that any generated message *must* apply the same message identifier if invoked more-then-once to have deduplication working on the infrastructure level or to solve this at the logical level by validating message data. [Read more on how to set a message identifier.](/nservicebus/messaging/message-identity.md)
+
+
+## Concurrency
+
+Outbox cannot guarantee that handlers or sagas are invoked exactly once. Outbox does not prevent concurrent attempts to process the same message but ensures that the outcome of the processing is persisted only once for resources that share the outbox transaction.
+
+The following scenarios might cause handlers to be invoked multiple times:
+
+- Sender sends multiple physical message with the same message identifier (submit order form)
+- Transport can send multiple messages when having a failover, they operate in a at-least-once delivery model
+- The outbox dispatch stage can be invoked more-than-once if the previous dispatch (partially) failed and is retried.
+
+In those scenarios, multiple physical copies of the same logical message exist in the queue. If an endpoint allows concurrent message processing or is scaled out it can happen that these multiple copies are processed concurrently.
+
+
+## Non transactional resources
+
+If non-transactional resources are being accessed then make sure processing is idempotent. Non-transactional resources are not guaranteed to be handled in an idempotent way by the Outbox. If more-than-once invocation is unwanted then this can be almost prevented completely by making sure only a single endpoint instance is running and have [configured the endpoint to process messages sequentially by setting the maximum concurrency to 1](/nservicebus/operations/tuning.md#tuning-concurrency). Only data modified via a shared storage session is made idempotent via the outbox and has exact-once semantics. See [persistence specific documention](/nservicebus/outbox/#Persistence) on how to obtain the shared storage session.
```

Comment:
> If more-than-once invocation is unwanted then this can be almost prevented completely by making sure only a single endpoint instance is running and have....

might be better as
> If more-than-once invocation is not wanted, then this can be prevented almost completely by making sure that only a single endpoint instance is running and having configured that endpoint with a maximum concurrency of 1.